File: api-node/healthcheck/healthcheck.js
const http = require("http");

const PORT = process.env.PORT || 3000;

const options = {
  host: "localhost",
  port: PORT,
  path: "/ping",
  timeout: 2000,
};

const req = http.request(options, (res) => {
  console.log("STATUS:", res.statusCode);
  process.exit(res.statusCode === 200 ? 0 : 1);
});

req.on("error", (err) => {
  console.error("ERROR:", err.message);
  process.exit(1);
});

req.end();

File: api-golang/healthcheck/healthcheck.go
package main

import (
	"fmt"
	"net/http"
	"os"
	"time"
)

func main() {
	port := os.Getenv("PORT")
	if port == "" {
		port = "8080"
	}

	client := http.Client{
		Timeout: 2 * time.Second,
	}

	resp, err := client.Get("http://localhost:" + port + "/ping")
	if err != nil {
		fmt.Println("ERROR:", err)
		os.Exit(1)
	}
	defer resp.Body.Close()

	if resp.StatusCode == 200 {
		fmt.Println("OK")
		os.Exit(0)
	}

	fmt.Println("STATUS:", resp.StatusCode)
	os.Exit(1)
}
